mmv1/templates/terraform/pre_update/network_services_gateway.erb

```diff
@@ -0,0 +1,4 @@
+if d.Get("type") == "SECURE_WEB_GATEWAY" {

```

Comment:
Can you explain what this is doing and why it was added? From what I can tell, it was previously possible for a user to have a `google_network_services_gateway` with `type = "SECURE_WEB_GATEWAY"`, and then change a field like `server_tls_policy` or `description`, which would trigger this condition, and presumably produce different results than before. Why do these 2 fields need to be set, and what happened before when they weren't set?

Reply:
I was just double checking with a terraform code using the actual provider and when trying to update the **description** of `google_network_services_gateway ` with `type = SECURE_WEB_GATEWAY` it throw the following error:
`Error: Error waiting for Updating Gateway: Error code 13, message: an internal error has occurred`
_The update send the **name**, the **ID** and the updated **description** fields only:_
```
# google_network_services_gateway.foobar will be updated in-place
  ~ resource "google_network_services_gateway" "foobar" {
      ~ description                          = "my description" -> "new description"
        id                                   = "projects/{PROJECT}/locations/us-east1/gateways/{NAME}"
        name                                 = {NAME}
        # (18 unchanged attributes hidden)
    }
```


When updating through **GCP console**, the request sends all the fields that the Gateway contains:
```
gateway: {
  addresses: [1]
  certificate_urls: [1]
  description: "new description"
  gateway_security_policy: {POLICY}
  name: "projects/{PROJECT}/locations/us-east1/gateways/{NAME}"
  network: {NETWORK}
  ports: [1]
  subnetwork: {SUBNETWORK}
  type: "SECURE_WEB_GATEWAY"
}
update_mask: {
  paths: [
    0: "certificate_urls"
    1: "description"
    2: "gateway_security_policy"
]}
```
_Also, **GCP Console** uses API v1beta1_


To update using the **API Explorer** first I used all fields and then I was removing one by one to check which fields are required and those 2 (`name` and `type`) were required to update the Gateway through **API Explorer**
```
PATCH https://networkservices.googleapis.com/v1/projects/{PROJECT}/locations/us-east1/gateways/{NAME}?updateMask=description&key=[YOUR_API_KEY] HTTP/1.1

Authorization: Bearer [YOUR_ACCESS_TOKEN]
Accept: application/json
Content-Type: application/json

{
  "description": "new description",
  "name": "{NAME}",
  "type": "SECURE_WEB_GATEWAY"
}
```

**_Conclusion:_ Since GCP Console sends all fields it's able to update the gateway. So this modification that I did will send the required fields (name and type) to be able to update as well.**

Reply:
Hmm ok, that's interesting behavior, but seems to match your logic. I suppose it's safer to stick with the condition you've included, but I do wonder if it might be cleaner to _always_ include name and type in the request.

Reply:
At first I did that... But if the `type` is other than `SECURE_WEB_GATEWAY ` the request fail.
_That broke some tests when I was starting the development._

Reply:
@roaks3 regarding the **fields description**, it's still tracked on our radar but it was was deprioritized together with ngprabhu@

Reply:
Awesome, thanks for those answers!